### ISSUE TITLE
Question for InfiniteScrolling: Horizontal Scrolling

### ISSUE THREAD

**Issue by Mike-E-angelo** [meziantou/Meziantou.Framework#511]
To start with, thank you very much for your efforts in this project, they are remarkable.  I have been looking into your InfiniteScroll control, and it appears to scroll vertically.  I am interested in a horizontal scroll control for a mobile scenario like what is seen here:

![](https://i.imgur.com/Wtdk7EH.gif)

I tried doing a simple display: flex on the parent div and am experiencing some interesting results.  Using the following code:

```
<div style="width: 512px;overflow-x:scroll; display: flex">
    <InfiniteScrolling ItemsProvider="@GetItems">
        <ItemTemplate>
            <div style="border: 1px solid blue; background: red; width: 100px; height: 100px">@context</div>
        </ItemTemplate>
        <LoadingTemplate>
            <div><span class="oi oi-reload spin"></span> Loading more items...</div>
        </LoadingTemplate>
    </InfiniteScrolling>
</div>

@code {
    Task<IEnumerable<int>> GetItems(InfiniteScrollingItemsProviderRequest request)
    {
        Debug.WriteLine("Requested! {0}", request.StartIndex);
        return Task.FromResult(Enumerable.Range(request.StartIndex, 10));
    }
}
```

I get the following:
![](https://i.imgur.com/FjpdGpG.gif)

Two notes:

1. I have to assign the width of the container directly to make this work.  Any % (e.g. `width: 100%`) or default (no) setting results in an infinite call to the ItemsProvider delegate.  It would be nice to know if a % could work with this scenario.
2. You can see that width is set to 100px but is not honored in the rendered output.  My CSS is not all that great so I am curious if this is something that is assigned by your side or if you are able to eyeball what is occurring here.  I will investigate further.

Thank you for any assistance you can provide,
Michael

**Comment by meziantou** [comment]
I haven't tested the code with a horizontal scroll, but I think the issue is that it doesn't find the scroll container. Maybe you need to handle `overflowX` instead of `overflowY` in https://github.com/meziantou/Meziantou.Framework/blob/main/src/Meziantou.AspNetCore.Components/wwwroot/InfiniteScrolling.ts#L30

**Comment by Mike-E-angelo** [comment]
Thank you for the reply @meziantou that is exactly what I landed on after poking/prodding for the past day or so.  Additionally, the [dimensions for the detector](https://github.com/meziantou/Meziantou.Framework/blob/main/src/Meziantou.AspNetCore.Components/InfiniteScrolling/InfiniteScrolling.razor#L20) should be changed to `style="width:1px"`.

One last note, probably more important.  When the list does reach the end, and there are no more elements to load, the `LoadMoreItems` call starts an infinite loop, constantly calling the source with the query.  I am not sure if you are aware of this, or if I had this configured wrong, but in the version I made, I dispose the control when such a condition occurs.  I thought I'd bring it up for your awareness.

**Comment by meziantou** [comment]
I've updated the code and the samples to demo the InfiniteScrolling component with a horizontal container. Also, it should not load items indefinitely anymore when you reach the end of the collection. Let me know if it works well.

**Comment by Mike-E-angelo** [comment]
Thank you very much for the improvements @meziantou.  I did try it out and it worked for me. 👍  The one remaining item that I see is that the current design will make another call to the source when no items remain.  In the case of the version I made (and with the Blazor `Virtualize` component), a property is defined for the total number of items in the query that helps to determine if another call needs to be made to the query source.
